config/src/main/java/com/typesafe/config/ConfigMemorySize.java

```diff
@@ -26,16 +29,43 @@ private ConfigMemorySize(long bytes) {
      * @param bytes a number of bytes
      * @return an instance representing the number of bytes
      */
-    public static ConfigMemorySize ofBytes(long bytes) {
+    public static ConfigMemorySize ofBytes(BigInteger bytes) {
         return new ConfigMemorySize(bytes);
     }
 
+    /**
+     * Constructs a ConfigMemorySize representing the given
+     * number of bytes.
+     * @param bytes a number of bytes
+     * @return an instance representing the number of bytes
+     */
+    public static ConfigMemorySize ofBytes(long bytes) {
+        return new ConfigMemorySize(BigInteger.valueOf(bytes));
+    }
+
     /**
      * Gets the size in bytes.
+     *
      * @since 1.3.0
      * @return how many bytes
      */
     public long toBytes() {
+        if (bytes.bitLength() < 64)
+            return bytes.longValue();
+        else
+            throw new IllegalArgumentException(
```

Comment:
ConfigException.BadValue is desirable here; but we'd probably have to make up a ConfigOrigin :-/ which might be uglier than it's worth. In any case documenting the throw is probably a good idea.

Reply:
thank you, I've documented the exception.